src/Umbraco.Core/Services/NotificationService.cs

```diff
@@ -204,6 +204,8 @@ public Notification CreateNotification(IUser user, IEntity entity, string action
             if (createSubject == null) throw new ArgumentNullException("createSubject");
             if (createBody == null) throw new ArgumentNullException("createBody");
 
+            allVersions = allVersions.OrderBy(x => x.UpdateDate).ToArray();
```

Comment:
I'll pull this in but would probably be better to do this order by before passing the allVersions parameter to this method.